### DISCUSSION TITLE
Maximum Stress Extraction

### DISCUSSION THREAD

**Discussion by vacupenda** [wenbinyugroup/.github#23] [VABS Related Discussions] [answered]
After performing stress recovery, how can I extract the maximum stress value, or which file contains this maximum stress value?

**Comment by unitasium** [comment]
The local stress data can be found in the following files:

- `<input_file_name>.S`: for each Gauss point measured in the cross-sectional coordinate system
- `<input_file_name>.SM`: for each Gauss point measured in the material coordinate system
- `<input_file_name>.SN`: for each node measured in the cross-sectional coordinate system
- `<input_file_name>.SMN`: for each node measured in the material coordinate system
- `<input_file_name>.ELE`: averaged for each element in both coordinate systems

Detailed arrangement of data in each file can be found in the VABS user manual.

**Reply by vacupenda** [reply]
Thank you for your explanation. Now I understand where the stress recovery components S11, S12, S13, S22, S23, and S33 come from, but how is the  "stress (global) - Full field" obtained?I would like to extract the maximum value of this data under different load conditions separately. Is there any way to do this?
![stress](https://github.com/user-attachments/assets/41d73a96-262d-4c85-9335-6af47cc3f00a)

**Reply by unitasium** [reply]
The "full field" view is created by importing the stress/strain tensors as a whole to gmsh. However, the plot on the right is generated by gmsh applying some postprocessing function to the tensors and has nothing to do with VABS.

What does "maximum stress value" mean in your case? Do you need the max values of each stress component, some kind of equivalent stress calculated using all components, or quantities like failure index/safety factor calculated based on certain failure criterion?

**Reply by vacupenda** [reply]
I need some kind of equivalent stress calculated using all components.

**Reply by unitasium** [reply]
Then you need to write your own code to read the stress file such as `ELE` and apply the equivalent stress function to each row.

**Reply by wenbinyu-PU** [reply, marked as answer]
If you want to use the stresses for failure/allowables check, then you can directly use VABS failure analysis, which is more convenient. For composites, to check the failure, some type of equivalent stress is not sufficient. You need all six stress components in the material coordinate systems and a well-established failure criterion.
